### PR TITLE
IGVF-336-crashfix

### PR DESCRIPTION
**Pull request by dougli1sqrd**
* Pulls in @forresttanaka mapprofile nextjs api
* The mapprofile api will take a schema name and return the associated collection
* This collection is then used for the Add button logic
* Removed the hard coded mapping between schema and collection

### CODE REVIEW

File: components/profile-map.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        })
+        .forEach((p) => {
+          const profileName = p["$id"].match(/^\/profiles\/(.+).json$/)[1];
+          // Get the mapped collection for the profile schema
+          api
+            .getObject(`/api/mapprofile?profile=${profileName}`)
+            .then((collect) => {
+              setMapping((current) => {
+                return { ...current, [profileName]: collect };
+              });
```
Comment (forresttanaka) [inline components/profile-map.js:30]: Might be a good place to use implicit return, but that’s completely optional.

File: pages/api/mapprofile.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            .json({ error: "Bad Request" });
+        }
+      });
+    } else {
+      res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({ error: "Bad Request" });
+    }
+  } else {
+    res.status(HTTP_STATUS_CODE.METHOD_NOT_ALLOWED).end();
+  }
+};
```
Comment (forresttanaka) [inline pages/api/mapprofile.js:27]: If you haven’t yet, make sure to write a ticket for that warning you get in the browser console.

**Review by forresttanaka — DISMISSED** [review]
Just a couple optional changes.

File: pages/api/mapprofile.js (hunk 2 above)
Comment (dougli1sqrd) [reply pages/api/mapprofile.js:27]: https://igvf.atlassian.net/browse/IGVF-338

**Comment by forresttanaka** [comment]
Sorry @dougli1sqrd; could you make one more change to this branch — not for a problem, but to make another part of the code simpler? In ProfileMapContext could you provide a _second_ property: `profiles`. In it, just put the result of retrieving /profiles. Then in your `ProfileMapContext.Provider` wrapper, provide that as the second property in the `value` object.

**Comment by forresttanaka** [comment]
@dougli1sqrd Thinking about it more, it makes sense to _not_ make that change in this branch since we’ll need another ticket to simplify the other code module, so we could just make the change I described in that branch.

**Review by forresttanaka — DISMISSED** [review]
Awesome!

**Pull request merged**